polkadot/node/subsystem-bench/src/lib/network.rs

```diff
@@ -476,25 +483,36 @@ impl EmulatedPeer {
 	pub async fn send_message(&mut self, message: NetworkMessage) {
 		self.tx_limiter.reap(message.size()).await;
 
-		if self.latency_ms == 0 {
-			self.to_node.unbounded_send(message).expect("Sending to the node never fails");
-		} else {
+		if self.has_latency() {
+			let latency = self.emulate_latency();
```

Comment:
nit: we could have a `get_latency()` which returns a `Option<tokio::time::Sleep>` instead of these two functions and use `if let Some(latency) = peer.get_latency() {...`